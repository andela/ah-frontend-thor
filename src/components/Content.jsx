import React from "react";

const Content = () => {
	return (
		<div className="container">
			<form>
				<div className="form-group">
					<input
						type="email"
						className="form-control"
						placeholder="Enter email"
					/>
				</div>
				<div className="form-group">
					<input
						type="password"
						className="form-control"
						placeholder="Enter Password"
					/>
				</div>
				<button type="submit" className="btn btn-primary">
					Login
				</button>
			</form>
		</div>
	);
};
export default Content;
