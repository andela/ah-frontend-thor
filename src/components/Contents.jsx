import React from "react";

import SocialButtons from "./login/SocialButtons";

const Content = () => (
  <div className="container">
    <SocialButtons />
  </div>
);

export default Content;
