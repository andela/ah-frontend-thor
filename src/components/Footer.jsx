import React from "react";

export default () => (
  <div>
    {/* <!-- Footer --> */}
    <footer className="page-footer font-small unique-color-dark">


      {/* <!-- Footer Elements --> */}
      <div className="container">

        {/* <!-- Grid row--> */}

        <div className="row">
          {/* <!-- Grid column --> */}
          <div className="col-md-12 py-5">
            <div className="mb-5 flex-center">

              {/* <!-- Facebook --> */}
              <a className="fb-ic">
                <i className="fa fa-facebook fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
              {/* <!-- Twitter --> */}
              <a className="tw-ic">
                <i className="fa fa-twitter fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
              {/* <!-- Google +--> */}
              <a className="gplus-ic">
                <i className="fa fa-google-plus fa-lg white-text mr-md-5 mr-3 fa-2x" />
              </a>
            </div>
          </div>
          {/* <!-- Grid column --> */}

        </div>
        {/* <!-- Grid row--> */}

      </div>
      {/* <!-- Footer Elements --> */}

      {/* <!-- Copyright --> */}
      <div className="footer-copyright text-center py-3">
        Â© 2018 Copyright:
        <a href="#">AuthorsHaven.com</a>
      </div>
      {/* <!-- Copyright --> */}

    </footer>

  </div>
);
