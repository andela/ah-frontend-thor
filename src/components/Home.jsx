import React from "react";

const Home = () => (
  <div>
    Author's Haven
  </div>
);
export default Home;
