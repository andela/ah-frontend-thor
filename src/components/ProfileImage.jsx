import React, { Component } from "react";
import { connect } from 'react-redux';
import { fetchProfile } from "../actions/profileAction";
import PropTypes from "prop-types";
export class ProfileImage extends Component {
  componentDidMount(){
    this.props.fetchProfile();
  }


  render() {
    const username = localStorage.getItem("username");
    if(username){
      return (<div className="avatar mx-auto">

      <img
        src={this.props.profile.user.image}
        className="header-profile"
        alt="image"
      />
    </div>);
    }else{
      return "";
    }
  }
}
ProfileImage.propTypes = {
  fetchProfile: PropTypes.func.isRequired
};
const mapStateToProps = state =>{
  return {
    profile: state.profile
  };
}
export default connect(mapStateToProps, {fetchProfile})(ProfileImage);
