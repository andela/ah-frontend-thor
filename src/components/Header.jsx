/* eslint-disable react/no-unescaped-entities */
import React from "react";
import { NavLink } from "react-router-dom";
import ProfileImage from "./ProfileImage";

const Header = () => (
  <header>
    <nav className="navbar fixed-top navbar-expand-lg navbar-dark unique-color-dark scrolling-navbar">
      <div className="container">
        <a className="navbar-brand" href="/index.html">
          <strong>Author's Haven</strong>
        </a>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav mr-auto">
            <li>
              <NavLink to="/post_article" className="navbar-brand">
                create Article
              </NavLink>
            </li>
          </ul>
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <form className="form-inline">
                <div className="md-form my-1">
                  <input
                    className="form-control mr-sm-2"
                    type="text"
                    placeholder="Search"
                    aria-label="Search"
                  />
                </div>
              </form>
            </li>
            <li>
              <NavLink to="/Profile" className="navbar-brand">
                <ProfileImage />
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </header>
);
export default Header;
