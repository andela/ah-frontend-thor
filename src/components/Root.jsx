/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React from "react";

import Header from "./Header";

const Root =({children})=> (
  <div>
    <Header />
    {children}
  </div>
);
export default Root;
