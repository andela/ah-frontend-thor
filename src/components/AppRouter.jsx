/* eslint-disable import/no-named-as-default */
/* eslint-disable react/jsx-indent */
import { Route, BrowserRouter as Router } from "react-router-dom";

import { Provider } from "react-redux";
import store from "../store";

import React from "react";
import Home from "./Home";
import Login from "../containers/Login";
import signUp from "./signup/SignUp";
import Root from "./Root";
import ForgotPasswordPage from "../containers/resetPassword/ForgotPasswordContainer";
import ChangePasswordPage from "../containers/resetPassword/ChangePasswordContainer";

import Profile from "../containers/Profile/Profile";
import EditProfile from "../containers/Profile/EditProfile";
import "../assets/css/styles.css";
import PostArticle from "./article/PostArticle";

const AppRouter = () => (
  <Provider store={store}>
    <Router>
      <React.Fragment>
        <Root>
          <div>
            <Route exact path="/" component={Home} />
            <Route path="/login" component={Login} />
            <Route path="/signup" component={signUp} />
            <Route path="/home" component={Home} />
            <Route path="/forgot_password" component={ForgotPasswordPage} />
            <Route path="/update_password" component={ChangePasswordPage} />
            <Route path="/profile" component={Profile} />
            <Route path="/editprofile" component={EditProfile} />
            <Route path="/post_article" component={PostArticle} />
          </div>
        </Root>
      </React.Fragment>
    </Router>
  </Provider>
);
export default AppRouter;
