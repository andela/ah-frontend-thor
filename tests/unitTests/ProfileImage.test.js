import React from "react";
import { shallow } from "enzyme";
import { ProfileImage } from "../../src/components/ProfileImage";

it('test profile image html', () => {
  const username = "kalyango"
  const component = shallow(
    <ProfileImage fetchProfile={jest.fn()} username={username} />
  );
  expect(component).toMatchSnapshot();
});
