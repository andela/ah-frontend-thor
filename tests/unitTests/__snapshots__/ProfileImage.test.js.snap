// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test profile image html 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileImage
    fetchProfile={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    username="kalyango"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "",
  Symbol(enzyme.__nodes__): Array [
    "",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
